src/PrinterStatus.php

```diff
@@ -0,0 +1,412 @@
+    }
+
+    /**
+     * @return bool
+     */
+    public function isMediaDieCut() : bool
+    {
+        return $this->getMediaBit(7) === 0;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isMediaContinuous() : bool
+    {
+        return $this->getMediaBit(7) === 1;
+    }
+
+    /**
+     * Returns true if the print head (lid) is open
+     *
+     * @return bool
+     */
+    public function isHeadUp() : bool
+    {
+        return !!$this->data[self::POS_HEAD_UP];
+    }
+
+    /**
+     * @return bool
+     */
+    public function isRibbonOut() : bool
+    {
+        return !!$this->data[self::POS_RIBBON_OUT];
+    }
+
+    /**
+     * @return bool
+     */
+    public function isThermalTransferMode() : bool
+    {
+        return !!$this->data[self::POS_THERMAL_TRANSFER_MODE] || $this->getMediaBit(0) === 1;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isDirectThermalMode() : bool
+    {
+        return !$this->data[self::POS_THERMAL_TRANSFER_MODE] && $this->getMediaBit(0) === 0;
+    }
+
+    /**
+     * Returns the current print mode. See `PrinterStatus::POS_PRINT_MODE`
+     *
+     * @return string
+     */
+    public function getPrintMode() : string
+    {
+        return self::PRINT_MODES[$this->data[self::POS_PRINT_MODE]];
+    }
+
+    /**
+     * ZPL Documentation doesn't really give much of a clue to what the return value means.
+     *
+     * @return int
+     */
+    public function getPrintWidthMode() : int
+    {
+        return (int) $this->data[self::POS_PRINT_WIDTH_MODE];
+    }
+
+    /**
+     * Returns true if label waiting in Peel-off Mode
+     *
+     * @return bool
+     */
+    public function isLabelWaiting() : bool
+    {
+        return !!$this->data[self::POS_LABEL_WAITING];
+    }
+
+    /**
+     * @return int
+     */
+    public function getLabelsRemainingCount() : int
+    {
+        return (int) $this->data[self::POS_LABELS_REMAINING];
+    }
+
+    /**
+     * Documentation states that this will always return true
+     *
+     * @return bool
+     */
+    public function isFormatWhilePrinting() : bool
+    {
+        return !!$this->data[self::POS_FORMAT_WHILE_PRINTING];
+    }
+
+    /**
+     * @return int
+     */
+    public function getImagesInMemoryCount() : int
+    {
+        return (int) $this->data[self::POS_IMAGES_IN_MEMORY];
+    }
+
+    /**
+     * Dear God, why?!
+     *
+     * @return string
+     */
+    public function getPassword() : string
+    {
+        return $this->data[self::POS_PASSWORD];
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasStaticRam() : bool
+    {
+        return !!$this->data[self::POS_STATIC_RAM_INSTALLED];
+    }
+
+    /**
+     * Returns the the value of a specific bit in the communications settings value
+     *
+     * @param int $bit Zero based binary position
+     * @return int
+     */
+    private function getCommBit(int $bit) : int
```

Comment:
I think this would be easier to follow if you used bitwise operators:

```
private function getCommBit(int $bit) : int
{
    $asInt = (int) $this->data[self::POS_COMM_SETTINGS];

    $mask = 1 << $bit;

    return $asInt & $mask;
}
```

Reply:
I knew there was a better bitwise way to do this. Thanks.

Reply:
Fixed in e154f07c7a09db20067bd832c02b91f7c0532a12